### DISCUSSION TITLE
Using podman-compose on FreeBSD

### DISCUSSION THREAD

**Discussion by VincentBentley** [containers/podman-compose#1055] [General]
I am experimenting with Podman on FreeBSD. It cannot be run rootless on FreeBSD yet so I wondered what is the best way to install podman-compose as root?

**Comment by VincentBentley** [comment]
I guess a FreeBSD pkg could be made to install podman-compose on the system once all of the dependencies have been isolated. A ZFS diff between before and after snapshots makes this easier to track down all of the changes. I am unable to make a commitment to packaging podman-compose for FreeBSD at this time so I will do the following instead.

I will install podman-compose with pip into an unprivileged user's virtual environment. I will create an unprivileged user ID with the name 'rootless' which will give me a consistent pathname for invoking the Python interpreter in a virtual environment with sudo and without sudo from cron.

If all my production pip installs are done with the user ID 'rootless', and the virtual environment is created in that user's 'home' directory, the virtual environment's Python ./venv/bin/python would actually be /rootless/venv/bin/python . Obviously the 'rootless' user would NOT have sudo privileges, but other sudo users and root would be able to execute Python from that absolute pathname. E.g. `$ sudo /rootless/venv/bin/python`

A shell script /usr/local/sbin/podman-compose could invoke the real podman-compose . Elevated privileges would be required to run the shell script and with it, podman-compose.

A crontab script for user 'rootless' could report on pip updates and optionally install them into the user's virtual environment.

**Comment by jokester** [comment]
I was able to pkg install this: https://www.freshports.org/sysutils/podman-compose/
